### ISSUE TITLE
"Create (new) Groups"

### ISSUE THREAD

**Issue by AcroDirki** [OpenAero/main#220]
Hi guys!

During a training week we propsed 21 figures for 3 Unknowns, containing 3 different stall turns, 3 dfferent tailsides and 4 rolling circles. They were proposed in their families one after the other.
When the "create groups" button was clicked, it was noted that the overall order of the 21 figures was shifted, but the sequential / suborder of figures was maintained, concluding such that all 4 rolling circles had to be flown in a single unknown, resp. all stallturns / tailslides in a unknown. This didn't change with multiple executions of "create new groups". Hence, the algorithm is not truly random. It needs to be improved, either by a constraint, spreading the figures of family 2, 5 & 6 as well as spins and flicks over all unknowns or by true random function and multiple execution (which is may be the easier but not the better solution).

The algorithm, however, should be explained in more detail in the manual to make it transparent.

It would be appreciated very much, if this could be implemented before July (german glider nationals) as a final test before the world championship.

Kind regards,
D.

**Comment by OpenAero** [comment]
As the manual indicates the algorithm creates groups that have the best possible match in speed/attitude. It certainly is not random! Creating random groups would give a higher chance of impossible combinations, exactly what this algorithm wants to prevent.
Forcing the spread of families 2, 5 & 6 could be fairly easily implemented. Do you consider this to be the best solution?

**Comment by AcroDirki** [comment]
Well, this solution is a first attempt, which sounds convenient. 2, 5, 6 the spins and the flicks should be distributed to ensure a certain  versatility in every unknown. As I described above, a single unknown that contains 4 different rolling circles would not be appreciated by the competitors / judges etc, while in all other unknowns a rolling circle is missing.
However, there should be a certain random component in the creation of groups, which is finally constrained by the sequence feasibility (not more than 2 additionals needed), in order not to establish any kind of selection preference in terms of speed (i.e. when the first unknown contains low speed entries/exits only, resp. as a majority due to any algorithm, as some competing aircrafts are more or less convenient for them). It's about the neitrality of figure selection for the unknowns, keeping in mind that not in every championship all unknowns are flown pending on weather conditions...

**Comment by OpenAero** [comment]
After a complete rewrite of the routine, there is now a new algorithm available on https://openaero.net/devel . Please test thoroughly with as many combinations as possible!

It works as follows:
1) Label every figure with a Superfamily number (as in old Free sequences) which marks it as spin, flick, turn, hammerhead, tailslide or other
2) Randomize figure order, but subsequently sort by Superfamily
3) Figure by figure, assign to one of the proposals. Take into account distribution of Superfamilies and best fit with existing proposal figure speeds

However, because of the nature of this method of figure selection, it can not be guaranteed that there exists a solution that also accounts for a maximum of 2 additionals in each group and total K in limits.
In these cases it is up to the jury to find the best solution.

**Comment by OpenAero** [comment]
Thanks for the testing.

Regarding point 1), I tried this on a Windows 10 laptop with chrome 66 and 67. Both worked correctly. So I cannot reproduce. I will try to find out what may cause this for you but maybe there is some unusual setting on your PC?

Regarding the combinations of the amendment. Does a new run provide better combinations?
By definition, as these rules are so convoluted and plainly awkward I think creating a "perfect" algorithm would be very time consuming. It seems always a new issue pops up regarding the feasibility of groups. Therefore I regard this algorithm to be sufficient, with the responsibility for final group creation resting with the jury.

I will look at items 3) and 5) and the possibility of adding "Total K per group" to the group creation dialog.

**Comment by OpenAero** [comment]
Thinking about 1) some more: so your desktop has both a mouse and a touch screen? That may be where the problem originates.

**Comment by OpenAero** [comment]
Please check if the latest version on openaero.net/devel solves the "wild" grid info issue. I have made an update that works well on touch and regular devices but do not have the combination.

**Comment by AcroDirki** [comment]
Hi Ringo,

The "wild" grid box seems to be successfully catched :) in the actual devel-version.
I can´t move it at all now, but this doesn´t really disturb me as I still can hide it, if necessary.

Yes, I´ve got an Lenove AIO MZ910 which I can operate by Mouse and Keyboard as well as through the touchscreen. Honestly I´m quite new to Win10, so I have no conclusion, whether I have set a "strange" option somewhere. It´s most probably the default option, then.

The implementation in the grid box to sort the figures looks very convenient and works well.

Kind regards,
D.

**Comment by OpenAero** [comment]
Hi Dirk,

I had to make some more changes to the Grid Info box because of other
issues. Can you confirm that the "wild" box issue has not returned when
using openaero.net/devel on your Lenovo?

Regards,
Ringo

Op vr 22 jun. 2018 om 12:35 schreef AcroDirki <notifications@github.com>:

**Comment by OpenAero** [comment]
Completed. No recent developments.
